test/parallel/test-async-wrap-GH13045.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+const common = require('../common');
+
+// Checks for the issue in Refs.
+// An HTTP Agent reuses a TLSSocket, and makes a failed call to `asyncReset`.
+
+const assert = require('assert');
+const https = require('https');
+const fs = require('fs');
+
+// arrow factory so fail point will appear in the stack
+function mustNotErr() {
```

Comment:
Tried it, then there is no frame for the line with the failing assert (Actually the factory is not a solution I need explicit `(err) => assert.fail(err)`)

Generated error by setting `port: port + 1`
with `(err) => assert.fail(err)`
```js
assert.js:92
  throw new AssertionError({
  ^

AssertionError [ERR_ASSERTION]: Error: write EPROTO 101057795:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:openssl\ssl\s23_clnt.c:794:

    at ClientRequest.req.on (D:\code\node-cur\test\parallel\test-async-wrap-GH13045.js:63:35)
    at emitOne (events.js:115:13)
    at ClientRequest.emit (events.js:210:7)
    at TLSSocket.socketErrorListener (_http_client.js:397:9)
    at emitOne (events.js:115:13)
    at TLSSocket.emit (events.js:210:7)
    at onwriteError (_stream_writable.js:359:10)
    at onwrite (_stream_writable.js:377:5)
    at fireErrorCallbacks (net.js:522:13)
    at TLSSocket.Socket._destroy (net.js:563:3)
```
with just `assert.fail`
```js
assert.js:92
  throw new AssertionError({
  ^

AssertionError [ERR_ASSERTION]: Error: write EPROTO 101057795:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:openssl\ssl\s23_clnt.c:794:

    at emitOne (events.js:115:13)
    at ClientRequest.emit (events.js:210:7)
    at TLSSocket.socketErrorListener (_http_client.js:397:9)
    at emitOne (events.js:115:13)
    at TLSSocket.emit (events.js:210:7)
    at onwriteError (_stream_writable.js:359:10)
    at onwrite (_stream_writable.js:377:5)
    at fireErrorCallbacks (net.js:522:13)
    at TLSSocket.Socket._destroy (net.js:563:3)
    at WriteWrap.afterWrite [as oncomplete] (net.js:857:10)
```